sdk/keyvault/keyvault-keys/src/index.ts

```diff
@@ -148,6 +147,86 @@ export {
   logger
 };
 
+/**
+ * Formats the received options into RequestOptions
+ */
+function toRequestOptionsBase(options?: CreateKeyOptions) {
+  let expandedOptions: RequestOptionsBase = {};
+
+  if (options) {
+    const requestOptions = operationOptionsToRequestOptionsBase(options);
+    const {
+      enabled,
+      notBefore,
+      expiresOn: expires,
+      hardwareProtected: hsm,
+      ...remainingOptions
+    } = requestOptions;
+    expandedOptions = {
+      ...remainingOptions,
+      keyAttributes: {
+        enabled,
+        notBefore,
+        expires,
+        hsm
+      }
+    };
+  }
+
+  return expandedOptions;
+}
+
+/**
+ * Prepares the request options and generates a tracer span.
+ * @param name Name of the tracer span.
+ * @param reqOptions Request options.
+ */
+function prepareTrace(name: string, reqOptions: RequestOptionsBase) {
+  const tracer = getTracer();
+  const span = tracer.startSpan(name, reqOptions && reqOptions.spanOptions);
+  span.setAttribute("az.namespace", "Microsoft.KeyVault");
```

Comment:
I took this from existing code, but I wonder if these methods could be simplified.

Reply:
if you can think of some good way to not mutate the passed in options, create the span, and also return the right set of options for child spans (e.g. the core-http span that happens from the generated client), I'd love to see the approach.

Reply:
this is a common problem in many libraries. I do think we can do something to provide better ways of returning the passed-in options, though I think we'd rather standardize around OperationOptions since RequestOptionsBase is a dead interface

Reply:
@xirzec ohhh, let's study that approach then! We could clean everything up 😮